modules/objdetect/include/opencv2/objdetect/aruco_board.hpp

```diff
@@ -0,0 +1,408 @@
+     * as objPoints)
+     *
+     * Recommended way to set ids vector, which will fail if the size of ids does not match size
+     * of objPoints.
+     */
+    CV_WRAP void setIds(InputArray ids);
+
+    /** @brief change id for ids[index]
+     * @param index - element index in ids
+     * @param newId - new value for ids[index], should be less than Dictionary size
+     */
+    CV_WRAP void changeId(int index, int newId);
+    /** @brief return ids
+     */
+    CV_WRAP const std::vector<int>& getIds() const;
+
+    /** @brief set dictionary
+     */
+    CV_WRAP void setDictionary(const Ptr<Dictionary> &dictionary);
+
+    /** @brief return dictionary
+     */
+    CV_WRAP Ptr<Dictionary> getDictionary() const;
+
+    /** @brief set objPoints
+     */
+    CV_WRAP void setObjPoints(const std::vector<std::vector<Point3f> > &objPoints);
+
+    /** @brief get objPoints
+     */
+    CV_WRAP const std::vector<std::vector<Point3f> >& getObjPoints() const;
+
+    /** @brief get rightBottomBorder
+     */
+    CV_WRAP const Point3f& getRightBottomBorder() const;
+
+protected:
+    /** @brief array of object points of all the marker corners in the board each marker include its 4 corners in this order:
+     * -   objPoints[i][0] - left-top point of i-th marker
+     * -   objPoints[i][1] - right-top point of i-th marker
+     * -   objPoints[i][2] - right-bottom point of i-th marker
+     * -   objPoints[i][3] - left-bottom point of i-th marker
+     *
+     * Markers are placed in a certain order - row by row, left to right in every row.
+     * For M markers, the size is Mx4.
+     */
+    CV_PROP std::vector<std::vector<Point3f> > objPoints;
+
+    /// the dictionary of markers employed for this board
+    CV_PROP Ptr<Dictionary> dictionary;
+
+    /// coordinate of the bottom right corner of the board, is set when calling the function create()
+    CV_PROP Point3f rightBottomBorder;
+
+    /** @brief vector of the identifiers of the markers in the board (same size than objPoints)
+     * The identifiers refers to the board dictionary
+     */
+    CV_PROP_RW std::vector<int> ids;
+};
+
+/**
+ * @brief Draw a planar board
+ * @sa drawPlanarBoard
+ *
+ * @param board layout of the board that will be drawn. The board should be planar,
+ * z coordinate is ignored
+ * @param outSize size of the output image in pixels.
+ * @param img output image with the board. The size of this image will be outSize
+ * and the board will be on the center, keeping the board proportions.
+ * @param marginSize minimum margins (in pixels) of the board in the output image
+ * @param borderBits width of the marker borders.
+ *
+ * This function return the image of a planar board, ready to be printed. It assumes
+ * the Board layout specified is planar by ignoring the z coordinates of the object points.
+ */
+CV_EXPORTS_W void drawPlanarBoard(const Ptr<Board> &board, Size outSize, OutputArray img,
```

Comment:
The function should stay in contrib as soon as there is Board::draw method in the new api.